Networking/Sources/Networking/Networking.swift

```diff
@@ -1,6 +1,36 @@
+import Alamofire
+import Foundation
+
 public struct Networking {
-    public private(set) var text = "Hello, World!"
+    static let shared = Networking()
 
     public init() {
     }
+    
+    public func fetchAllProducts<T: Codable>(expectingReturnType: T.Type, completion: @escaping ((Result<T?,Error>) -> Void)){
```

Comment:
I think you may not need to pass the type since you can specify it in the return type.

Reply:
Why you need Codable? `Decodable` is enough. Let's discuss it if needed.